code/Pagetypes/FreshdeskPage.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+
+class FreshdeskPage extends Page
+{
+}
+
+class FreshdeskPage_Controller extends Page_Controller
+{
+    /**
+     * Priority mapping for Freshdesk tickets.
+     * @var array
+     */
+    private static $freshdeskPriority = [
+        0 => 'any',
+        1 => 'low',
+        2 => 'medium',
+        3 => 'high',
+        4 => 'urgent',
+    ];
+
+    private static $dependencies = [
+        'freshdesk' => '%$FreshdeskAPI',
+    ];
+
+    /**
+     * @var FreshdeskAPI
+     */
+    public $freshdesk;
+
+    /**
+     * @var array
+     */
+    private static $_tickets = [];
+
+    /**
+    * Returns Freshdesk tickets for a user
+    *
+    * @return PaginatedList
+    */
+    public function getTickets()
+    {
+        $currentMember = \Member::currentUser();
+        if (!$currentMember || !$currentMember->exists()) {
+            return \Security::permissionFailure();
+        }
+
+        if (!self::$_tickets) {
+            self::$_tickets = $this->freshdesk->getUserTickets($currentMember);
+        }
+
+        $tickets = self::$_tickets;
+
+        if (empty($tickets) || count($tickets) == 0) {
+            return false;
+        }
+
+        // Get all open by default
+        $filter = ['status' => 2];
+        if ($this->request->getVar('status')) {
+            $filter['status'] = (int) $this->request->getVar('status');
+        }
+        if ($this->request->getVar('priority')) {
+            $filter['priority'] = (int) $this->request->getVar('priority');
+        }
+
+        $tickets = $this->filterTickets($tickets, $filter);
+        $tickets = $this->freshdesk->humanReadable($tickets, self::$freshdeskPriority);
+
+        return new PaginatedList($tickets, $this->request);
+    }
+
+    /**
+    * Filter tickets based on productID and status/priority filters
+    *
+    * @param array $tickets
+    * @return ArrayList $tickets
+    */
+    public function filterTickets($tickets, $filter)
```

Comment:
I don't think this function should be part of the public API. Looks like a private helper, so it would preferably be set `private`. When writing unit tests, you should only to test the **public** API of the unit (class in this case). This is so that you can change the internal implementation, and not having to rewrite tests. So you should be testing `getTickets`.